src/System.Threading/tests/MutexTests.cs

```diff
@@ -130,34 +126,47 @@ public void OpenExisting_NameUsedByOtherSynchronizationPrimitive_Windows()
             }
         }
 
-        [Fact]
-        public void AbandonExisting()
+        public static IEnumerable<object[]> AbandonExisting_MemberData()
         {
-            using (Mutex m = new Mutex())
+            var nameGuidStr = Guid.NewGuid().ToString("N");
+            for (int waitType = 0; waitType < 2; ++waitType) // 0 == WaitOne, 1 == WaitAny
             {
-                Task t = Task.Factory.StartNew(() =>
+                yield return new object[] { null, waitType };
+                foreach (var namePrefix in new[] { string.Empty, "Local\\", "Global\\" })
                 {
-                    Assert.True(m.WaitOne(FailedWaitTimeout));
-                // don't release the mutex; abandon it on this thread
-            }, CancellationToken.None, TaskCreationOptions.LongRunning, TaskScheduler.Default);
-                t.Wait();
-                Assert.Throws<AbandonedMutexException>(() => m.WaitOne(FailedWaitTimeout));
+                    yield return new object[] { namePrefix + nameGuidStr, waitType };
+                }
             }
+        }
 
-            using (Mutex m = new Mutex())
+        [Theory]
+        [MemberData(nameof(AbandonExisting_MemberData))]
+        public void AbandonExisting(string name, int waitType)
+        {
+            using (var m = new Mutex(false, name))
             {
                 Task t = Task.Factory.StartNew(() =>
                 {
                     Assert.True(m.WaitOne(FailedWaitTimeout));
+                    // don't release the mutex; abandon it on this thread
                 }, CancellationToken.None, TaskCreationOptions.LongRunning, TaskScheduler.Default);
                 t.Wait();
-                AbandonedMutexException ame = Assert.Throws<AbandonedMutexException>(() => WaitHandle.WaitAny(new[] { m }, FailedWaitTimeout));
-                Assert.Equal(0, ame.MutexIndex);
-                Assert.Equal(m, ame.Mutex);
+
+                switch (waitType)
+                {
+                    case 0: // WaitOne
+                        Assert.Throws<AbandonedMutexException>(() => m.WaitOne(FailedWaitTimeout));
+                        break;
+
+                    case 1: // WaitAny
+                        AbandonedMutexException ame = Assert.Throws<AbandonedMutexException>(() => WaitHandle.WaitAny(new[] { m }, FailedWaitTimeout));
+                        Assert.Equal(0, ame.MutexIndex);
+                        Assert.Equal(m, ame.Mutex);
```

Comment:
Do these asserts apply to case 0 as well?  If so, they could be moved to after the switch block.